src/BugsnagUnity/Native/Android/NativeInterface.cs

```diff
@@ -454,116 +445,89 @@ private Breadcrumb ConvertToBreadcrumb(IntPtr javaBreadcrumb)
-      var entries = AndroidJNI.CallObjectMethod(javaMetadata, MapEntrySet, new jvalue[]{});
-      var iterator = AndroidJNI.CallObjectMethod(entries, CollectionIterator, new jvalue[]{});
+      IntPtr javaMetadata = AndroidJNI.CallObjectMethod(javaBreadcrumb, BreadcrumbGetMetadata, new jvalue[]{});
+      IntPtr metadataEntries = AndroidJNI.CallObjectMethod(javaMetadata, MapEntrySet, new jvalue[]{});
+      AndroidJNI.DeleteLocalRef(javaMetadata);
+      IntPtr metadataIterator = AndroidJNI.CallObjectMethod(metadataEntries, CollectionIterator, new jvalue[]{});
+      AndroidJNI.DeleteLocalRef(metadataEntries);
 
-      while (AndroidJNI.CallBooleanMethod(iterator, IteratorHasNext, new jvalue[]{}))
+      while (AndroidJNI.CallBooleanMethod(metadataIterator, IteratorHasNext, new jvalue[]{}))
       {
-        var entry = AndroidJNI.CallObjectMethod(iterator, IteratorNext, new jvalue[]{});
-        var key = AndroidJNI.CallObjectMethod(entry, MapEntryGetKey, new jvalue[]{});
-        var value = AndroidJNI.CallObjectMethod(entry, MapEntryGetValue, new jvalue[]{});
-        if (key != IntPtr.Zero && value != IntPtr.Zero)
-        {
-          metadata.Add(AndroidJNI.GetStringUTFChars(key), AndroidJNI.GetStringUTFChars(value));
-        }
-        AndroidJNI.DeleteLocalRef(key);
-        AndroidJNI.DeleteLocalRef(value);
+        IntPtr entry = AndroidJNI.CallObjectMethod(metadataIterator, IteratorNext, new jvalue[]{});
+        string key = AndroidJNI.CallStringMethod(entry, MapEntryGetKey, new jvalue[]{});
+        string value = AndroidJNI.CallStringMethod(entry, MapEntryGetValue, new jvalue[]{});
         AndroidJNI.DeleteLocalRef(entry);
+
+        //Currently this means that Unity on Android does not support complex metadata for
+        //breadcrumbs
+        metadata.Add(key, value);
       }
 
-      AndroidJNI.DeleteLocalRef(javaMetadata);
-      AndroidJNI.DeleteLocalRef(entries);
-      AndroidJNI.DeleteLocalRef(iterator);
+      AndroidJNI.DeleteLocalRef(metadataIterator);
 
-      var type = AndroidJNI.CallObjectMethod(javaBreadcrumb, BreadcrumbGetType, new jvalue[]{});
-      var typeName = AndroidJNI.CallStringMethod(type, ObjectToString, new jvalue[]{});
+      IntPtr type = AndroidJNI.CallObjectMethod(javaBreadcrumb, BreadcrumbGetType, new jvalue[]{});
+      string typeName = AndroidJNI.CallStringMethod(type, ObjectToString, new jvalue[]{});
       AndroidJNI.DeleteLocalRef(type);
-      var name = "<empty>";
-      var nameObj = AndroidJNI.CallObjectMethod(javaBreadcrumb, BreadcrumbGetName, new jvalue[]{});
-      if (nameObj != IntPtr.Zero)
-      {
-        name = AndroidJNI.GetStringUTFChars(nameObj);
-      }
-      var timestamp = AndroidJNI.CallStringMethod(javaBreadcrumb, BreadcrumbGetTimestamp, new jvalue[]{});
+
+      string name = AndroidJNI.CallStringMethod(javaBreadcrumb, BreadcrumbGetName, new jvalue[]{});
+      if(name == null) name = "<empty>";
```

Comment:
Code style is currently not enforced with linters on this repo, but the convention is to use braces for single-line conditionals